src/lib/WalletInfo.ts

```diff
@@ -26,16 +32,44 @@ export class WalletInfo {
                        public fileExported: boolean = false,
                        public wordsExported: boolean = false) {}
 
+    public findContractByAddress(address: Nimiq.Address): ContractInfo | undefined {
+        return this.contracts.find((contract) => contract.address.equals(address));
+    }
+
+    public findContractsByOwner(address: Nimiq.Address): ContractInfo[] {
+        return this.contracts.filter((contract) => {
+            switch (contract.type) {
+                case Nimiq.Account.Type.VESTING: return (contract as VestingContractInfo).owner.equals(address);
+                case Nimiq.Account.Type.HTLC:
+                    return (contract as HashedTimeLockedContractInfo).sender.equals(address)
+                        || (contract as HashedTimeLockedContractInfo).recipient.equals(address);
+                default: return false;
+            }
+        });
+    }
+
+    public setContract(updatedContract: ContractInfo) {
+        const index = this.contracts.findIndex((contract) => contract.address.equals(updatedContract.address));
+        if (index < 0) {
+            // Is new contract
+            this.contracts.push(updatedContract);
+            return;
+        }
+
+        this.contracts.splice(index, 1, updatedContract);
```

Comment:
```suggestion
        this.contracts[index] = updatedContract;
```

Reply:
https://vuejs.org/2016/02/06/common-gotchas/#Why-isn%E2%80%99t-the-DOM-updating
> When you modify an Array by directly setting an index (e.g. arr[0] = val) or modifying its length property. Similarly, Vue.js cannot pickup these changes. Always modify arrays by using an Array instance method, or replacing it entirely. [...]